### PR TITLE
Add new authentication methods and verifications

### PR DESCRIPTION
**Pull request by Rochius**
I added new authentication methods and verifications

### CODE REVIEW

File: libraries/authentication/oauth2.py
Comment (DenisseC02) [inline libraries/authentication/oauth2.py:1]: Please remove the spaces from the lines 11, 16 and 20

File: libraries/authentication/api_key.py
Comment (DenisseC02) [inline libraries/authentication/api_key.py:1]: Please remove the space from the line 17

Comment (Rochius) [reply libraries/authentication/api_key.py:1]: I already remove the spaces, thank you

File: libraries/authentication/oauth2.py
Comment (Rochius) [reply libraries/authentication/oauth2.py:1]: I already remove the spaces, thank you

**Review by DenisseC02 — APPROVED** [review]

**Review by MartinJAA — APPROVED** [review]

**Pull request merged**